spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -243,34 +243,88 @@
         )
       end
 
-      it 'logs a message with counts of various outcomes when the job completes' do
-        allow(InPersonEnrollment).to receive(:needs_usps_status_check).
-          and_return(pending_enrollments)
-        stub_request_proofing_results_with_responses(
-          request_passed_proofing_results_args,
-          request_in_progress_proofing_results_args,
-          { status: 500 },
-          request_failed_proofing_results_args,
-          request_expired_proofing_results_args,
-        )
+      context 'with a set of outcomes to log' do
+        let(:responses) do
+          [
+            request_passed_proofing_results_args,
+            request_in_progress_proofing_results_args,
+            { status: 500 },
+            request_failed_proofing_results_args,
+            request_expired_proofing_results_args,
+          ]
+        end
+
+        before do
+          allow(InPersonEnrollment).to receive(:needs_usps_status_check).
+            and_return(pending_enrollments)
+          stub_request_proofing_results_with_responses(*responses)
+        end
 
-        job.perform(Time.zone.now)
+        it 'logs a message with counts of various outcomes when the job completes' do
+          job.perform(Time.zone.now)
 
-        expect(job_analytics).to have_logged_event(
-          'GetUspsProofingResultsJob: Job completed',
-          duration_seconds: anything,
-          enrollments_checked: 5,
-          enrollments_errored: 1,
-          enrollments_expired: 1,
-          enrollments_failed: 1,
-          enrollments_in_progress: 1,
-          enrollments_passed: 1,
-        )
+          expect(job_analytics).to have_logged_event(
+            'GetUspsProofingResultsJob: Job completed',
+            duration_seconds: anything,
+            enrollments_checked: 5,
+            enrollments_errored: 1,
+            enrollments_expired: 1,
+            enrollments_failed: 1,
+            enrollments_in_progress: 1,
+            enrollments_passed: 1,
+          )
 
-        expect(
-          job_analytics.events['GetUspsProofingResultsJob: Job completed'].
-            first[:duration_seconds],
-        ).to be >= 0.0
+          expect(
+            job_analytics.events['GetUspsProofingResultsJob: Job completed'].
+              first[:duration_seconds],
+          ).to be >= 0.0
+        end
+
+        it 'logs a message with counts of various outcomes when the job is terminated' do
+          error_message = 'too many!'
+          analytics_title = 'GetUspsProofingResultsJob: Job terminated'
+          # make a subclass of GetUspsProofingResultsJob that raises a
+          # ConcurrencyExceededError on its last enrollment
+          class ErrorJob < GetUspsProofingResultsJob
```

Comment:
because ruby blocks are weird, this is going to create `ErrorJob` at the top of the global namespace. (so no other tests can ever define an `ErrorJob` because they'd collide)

I would consider using the anonymous class syntax via a `let`:

```ruby
let(:error_job_class) do
  Class.new(GetUspsProofingResultsJob) do
    private

    def check_enrollment(...)
    # ...
    end
  end.tap do |klass|
    # define method using this block syntax so we can capture the outer `job_analytics` and not have to stub any instance
    klass.define_method(:analytics) { job_analytics }
  end
end
```

Reply:
Thank you, I did this in 5dc1542fdc06eac4ce8be7971cf1d66ab332bcc7

I ended up having to stub the `analytics` method anyway because variables I passed into the class definition lost their context (it'd throw `NameError: undefined local variable or method 'job_analytics'`)